packages/stream_chat/lib/src/api/channel.dart

```diff
@@ -1191,7 +1191,7 @@ class Channel {
 
   /// Call this method to dispose the channel client
   void dispose() {
-    state.dispose();
+    state!.dispose();
```

Comment:
this should be `state?.dispose()` as we can't be sure if we have already initialized state or not.

Reply:
or maybe add that assertion at the start, which we discussed